Restructure quickstart (#145)
precice/tutorials: quickstart/README.md

    - For example, [download](https://github.com/precice/precice/releases/latest) and install our binary package for Ubuntu 20.04 (Focal Fossa) by clicking on it or using the following commands:
    ```shell
    wget https://github.com/precice/precice/releases/download/v2.2.0/libprecice2_2.2.0_focal.deb
    sudo apt install ./libprecice2_2.2.0_focal.deb
    ```
   - If you prefer to try everything in an isolated environment, you may prefer using our [demo virtual machine](installation-vm.html).
3. We will use OpenFOAM here and in many of our tutorial cases, so [install OpenFOAM](adapter-openfoam-support.html) (most compatible version: latest ESI/OpenFOAM.com).
4. Download and install the [OpenFOAM-preCICE adapter](adapter-openfoam-get.html).
5. Couple OpenFOAM with a C++ rigid body solver. [Find the case in our tutorials](https://github.com/precice/tutorials/quickstart) and keep reading. You can either `git clone` the [tutorials repository](https://github.com/precice/tutorials), or [directly download the current state](https://github.com/precice/tutorials/archive/master.zip).

> ## What's next?

Reviewer: I just noticed that this section also needs some updates:
- Move the tutorials link from the wiki to the website
- Use the new YouTube channel link: https://www.youtube.com/c/preCICECoupling/
- Replace the "register to our workshop" link to a more general call to come to a workshop: https://www.precice.org/community.html

Author: Good catch (though you marked the wrong lines here).
I added now 'Meet our community and added the community link.

Reviewer: I marked the first line that GitHub allowed me to mark (the `## What's next`) :confused:

Author: Yes, I realized it too late since the line has not been changed. For the record: you can mark multiple lines as well by holding the 'plus' button.

Reviewer: The problem I faced is actually that the GitHub review tool does not let you mark a line that is not in a diff chunk affected by the PR. The `What's next?` was directly below the above changes, so it allowed me to mark this, but not the next lines.

2. Get and install preCICE. For Linux systems, this is pretty easy. Just pick what suits you best on [this overview page](installation-overview.html). Facing any problems? [Ask for help](community-channels.html).
    - For example, [download](https://github.com/precice/precice/releases/latest) and install our binary package for Ubuntu 20.04 (Focal Fossa) by clicking on it or using the following commands:
    ```shell
    wget https://github.com/precice/precice/releases/download/v2.2.0/libprecice2_2.2.0_focal.deb
    sudo apt install ./libprecice2_2.2.0_focal.deb

>     ```

Author: Maybe we should use placeholder, e.g. x.y.z for the version in order to avoid this problem in the future,

3. Build and couple two [C++ solverdummies](https://github.com/precice/precice/tree/master/examples/solverdummies/cpp).
4. You probably want to couple a solver you are already using, such as OpenFOAM. Since many of our tutorials use it, [install OpenFOAM](adapter-openfoam-support.html).
5. Download and install the [OpenFOAM-preCICE adapter](adapter-openfoam-get.html).

> 6. [Couple OpenFOAM with a c++ rigid body solver](https://github.com/precice/tutorials/quickstart).

Author: ... you mean the whole preCICE project, including adapters, tutorials and the documentation? :ok_hand:

> 6. [Couple OpenFOAM with a c++ rigid body solver](https://github.com/precice/tutorials/quickstart).

Reviewer: This is what we have with the VM. Here I am mainly commenting on "we need versioning in the tutorials". But that's for another time.

* Tell us [your story](community-projects.html).

## About the case

> This tutorial deals with a fluid-structure interaction problem. The fluid part of the simulation is computed using OpenFOAM and the rigid body motion is a rigid body model (implemented in C++) with only a single degree of freedom, namely the deflection angle of the flap in the channel. The rigid body is fixed in the origin at (0,0) and the force exerted by the fluid on the rigid body structure causes an oscillatory rotation of the body. The simulation runs for 2.5 seconds. In order to gain more control over the rigid body oscillation, a rotational spring is applied at the rigid body origin. After 1.5 seconds we increase the spring constant by a factor of 8 to stabilize the coupled problem. Feel free to modify these parameters (directly in `rigid_body_solver.cpp`) and increase the simulation time (in `precice-config.xml`).

Author: > A few minor comments, mostly very picky changes in the formulations and a rather important bug prevention in the scripts.
> 
> I ran the case and I observed that:
> 
>     * heavy_check_mark The case is running
> 
>     * x The `removeObsoleteFolders.sh` needs to remove the `cd Fluid` and the `cd ..`.
> 
>     * heavy_check_mark The `plotDisplacement.sh` works
> 
>     * heavy_check_mark ParaView works.
> 
>     * grey_question I still see a kind-of "explosion"  in the second half of the simulation (higher-frequency oscillation). Maybe we could add a note that this is expected and why.
> 
>     * x `clean.sh` needs a few fixes (see review)
> 
> 
> Looks very nice already! rocket

The `clean.sh` and `removeFolders` should be fine now. In case the following figure is "exploding" for you: that's intended. It should also be clear when reading through the introduction here.
![](https://user-images.githubusercontent.com/33414590/110659722-1ec89600-81c3-11eb-94a1-ef9cad784c54.png)

![overview](images/overview.png)

> ### Building the `rigid_body` participant

Reviewer: Technically, this is something independent of the coupling, so we should refer to this as "solver" or "code". I also think we could avoid the inline code (`rigid_body`), since this is not the exact name,

```suggestion
### Building the rigid body solver
```

> Before starting the coupled simulation, the `rigid_body` solver needs to be built using CMake. You can run the following commands from this directory to build the `rigid_body_solver.cpp`

Reviewer: Same here: 
```suggestion
Before starting the coupled simulation, the rigid body solver needs to be built using CMake. You can run the following commands from this directory to build the `rigid_body_solver.cpp`
```

```
cd solid-cpp && cmake . && make
```

### Running the coupled simulation

You may run the two simulations in two different terminals and watch their output on the screen using the `run.sh` scripts (or `run.sh -parallel`, option only available for OpenFOAM) located in each participant directory. You can cleanup the simulation using `clean.sh`.

> There is an [open issue](https://github.com/precice/openfoam-adapter/issues/26) that leads to additional "empty" result directories when running with some OpenFOAM versions, leading to inconveniences during post-processing. Please run the script `removeObsoleteSolvers.sh` to delete the additional files.

Author: I agree, it will just confuse newcomer.

Ini serial, the simulation takes roughly 30 seconds to compute.

### Visualizing the results

> You can visualize the simulation results of the `Fluid` participant using ParaView (use `paraFoam` to trigger the OpenFOAM native reader or load the (empty) file `Fluid.foam` into ParaView). The rigid body doesn't generate any readable output files, but the motion can be observed in the OpenFOAM data. In addition, one could visualize the coupling meshes including the exchanged coupling data. preCICE generates the relevant files during the simulation and stores them in the directory `coupling-meshes`. In order to visualize the results, load the VTK files in ParaView and apply a `Glyph` filter. Depending on the specific ParaView version, you might additionally need to disable the `ScaleArray` option by selecting `No scale array`, since the exchanged data might be inappropriate for a scaling operation. You can further add a `Warp By Vector` filter with `Displacements` to deform the coupling data.

Reviewer: Good catch renaming the variables in the `precice-config.xml`. We also need to update them here then:
```suggestion
You can visualize the simulation results of the `Fluid` participant using ParaView (use `paraFoam` to trigger the OpenFOAM native reader or load the (empty) file `Fluid.foam` into ParaView). The rigid body doesn't generate any readable output files, but the motion can be observed in the OpenFOAM data. In addition, one could visualize the coupling meshes including the exchanged coupling data. preCICE generates the relevant files during the simulation and stores them in the directory `coupling-meshes`. In order to visualize the results, load the VTK files in ParaView and apply a `Glyph` filter. Depending on the specific ParaView version, you might additionally need to disable the `ScaleArray` option by selecting `No scale array`, since the exchanged data might be inappropriate for a scaling operation. You can further add a `Warp By Vector` filter with `Displacement` to deform the coupling data.
```

(anywhere else?)

* Tell us [your story](community-projects.html).

## About the case

> This tutorial deals with a fluid-structure-interaction problem. The fluid part of the simulation is computed using OpenFOAM and the rigid body motion is a rigid body model written in c++ with only a single degree of freedom, namely the deflection angle of the flap in the channel. The rigid body is fixed in the origin at (0,0) and the force exerted by the fluid on the rigid body structure causes an oscillatory rotation of the body. The simulation runs for 2.5 seconds. In order to gain more control over the rigid body oscillation, a rotational spring is applied at the rigid body origin. After 1.5 seconds we increase the spring constant by a factor of 8 to stabilize the coupled problem. Feel free to modify these parameters and increase the simulation time.

Reviewer: ```suggestion
This tutorial deals with a fluid-structure interaction problem. The fluid part of the simulation is computed using OpenFOAM and the rigid body motion is a rigid body model (implemented in C++) with only a single degree of freedom, namely the deflection angle of the flap in the channel. The rigid body is fixed in the origin at (0,0) and the force exerted by the fluid on the rigid body structure causes an oscillatory rotation of the body. The simulation runs for 2.5 seconds. In order to gain more control over the rigid body oscillation, a rotational spring is applied at the rigid body origin. After 1.5 seconds we increase the spring constant by a factor of 8 to stabilize the coupled problem. Feel free to modify these parameters (directly in `rigid_body_solver.cpp`) and increase the simulation time (in `precice-config.xml`).
```

![overview](images/overview.png)

### Building the `rigid_body` participant

> Before starting the coupled simulation, the `rigid_body` solver needs to be compiled using `cmake`. You can run the following commands from this directory to build the `rigid_body` solver

Reviewer: ```suggestion
Before starting the coupled simulation, the `rigid_body` solver needs to be built using CMake. You can run the following commands from this directory to build the `rigid_body_solver.cpp`
```

```

> cd Solid && cmake . && make

Reviewer: ```suggestion
cd solid-cpp && cmake . && make
```

> ```

Author: I would still vote for a repository global `gitignore` as already discussed [here](https://github.com/precice/tutorials/pull/154#issuecomment-792590007), but if all of you are happy with case specific one, we need to add one in every directory,

> ```

Reviewer: Then you can leave it as it is and I will work on it later. Related issue: https://github.com/precice/tutorials/issues/87

### Running the coupled simulation

> You may run the two simulations in two different terminals and watch their output on the screen by using the `run.sh` scripts (or `run.sh -parallel` for OpenFOAM) located in each participant directory. You can cleanup the simulation using `clean.sh`.

Reviewer: ```suggestion
You may run the two simulations in two different terminals and watch their output on the screen using the `run.sh` scripts (or `run.sh -parallel`, option only available for OpenFOAM) located in each participant directory. You can cleanup the simulation using `clean.sh`.
```

There is an [open issue](https://github.com/precice/openfoam-adapter/issues/26) that leads to additional "empty" result directories when running with some OpenFOAM versions, leading to inconveniences during post-processing. Please run the script `removeObsoleteSolvers.sh` to delete the additional files.

Ini serial, the simulation takes roughly 30 seconds to compute.

### Visualizing the results

> You can visualize the simulation results of the `Fluid` participant using paraView (use paraFoam to trigger the OpenFOAM native reader or load the (empty) file `Fluid.foam` into paraView). The rigid body doesn't generate any readable output files, but the motion can be observed in the OpenFOAM data. In addition, one could visualize the coupling meshes including the exchanged coupling data. preCICE generates the relevant files during the simulation and stores them in the directory `coupling-meshes`. In order to visualize the results, load the `vtk` files in paraView and apply a `Glyph` filter. Depending on the specific paraView version, you might additionally need to disable the `ScaleArray` option by selecting `No scale array` since the exchanged data might be inappropriate for a scaling operation. You can further add a `Warp By Vector` filter with `Displacements` to deform the coupling data.

Reviewer: ```suggestion
You can visualize the simulation results of the `Fluid` participant using ParaView (use `paraFoam` to trigger the OpenFOAM native reader or load the (empty) file `Fluid.foam` into ParaView). The rigid body doesn't generate any readable output files, but the motion can be observed in the OpenFOAM data. In addition, one could visualize the coupling meshes including the exchanged coupling data. preCICE generates the relevant files during the simulation and stores them in the directory `coupling-meshes`. In order to visualize the results, load the VTK files in ParaView and apply a `Glyph` filter. Depending on the specific ParaView version, you might additionally need to disable the `ScaleArray` option by selecting `No scale array`, since the exchanged data might be inappropriate for a scaling operation. You can further add a `Warp By Vector` filter with `Displacements` to deform the coupling data.
```

> 

Reviewer: It would be nice to have a screenshot of ParaView with the options and the visualization here.

Author: Done

> {% include important.html content="We are currently working on a really nice getting started page. Till then, we hope these tips already help." %}

Reviewer: ```suggestion
```
This is the really nice getting started page ;-) We can always improve it further, but this disclaimer can be removed already.
